code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -96,11 +130,318 @@
+						holy_modifier += 2
+					else
+						if(turf_on.blessed) //The chaplain's spilt some of his holy water
+							holy_modifier += 1
+
+					if(prob(15*holy_modifier)) //Gotta have faith
+						to_chat (user, "<span class='notice'>By [bible.deity_name] it's working!.</span>")
+						unzombify()
+					else
+						to_chat (user, "<span class='notice'>Well, that didn't work.</span>")
+
+/mob/living/simple_animal/hostile/necro/zombie/turned/proc/unzombify()
+	if(host)
+		host.loc = get_turf(src)
+		if(!host.mind && src.mind) //This is assuming that, somehow, the host lost their soul, and it ended up in the zombie
+			mind.transfer_to(host)
+		host.resurrect() //It's a miracle!
+		host.revive()
+		visible_message("<span class='notice'>\The [src]'s eyes regain focus, the smell of decay vanishing, [host] has come back to their senses!.</span>")
+		host = null
+		qdel(src)
+	else
+		visible_message("<span class='notice'>\The [src] grumbles for a moment, then begins to decay at an accelerated rate, seems there was nobody left to save.</span>")
+		dust()
+
+/mob/living/simple_animal/hostile/necro/zombie/rotting
+	icon_living = "zombie_rotten"
+	icon_state = "zombie_rotten"
+	icon_dead = "zombie_rotten_dead"
+	desc = "A reanimated corpse that looks like it has seen better days. Whoever this was is long gone."
+	maxHealth = 100
+	health = 100
+	can_evolve = 1
+	break_doors = CAN
+
+/mob/living/simple_animal/hostile/necro/zombie/rotting/check_evolve()
+	..()
+	if(times_eaten > (1+times_revived)*2) //Have to have eaten at least twice and more than double that of times died
+		evolve(/mob/living/simple_animal/hostile/necro/zombie/putrid)
+	else if(times_revived > times_eaten+1) //Died at least twice
+		evolve(/mob/living/simple_animal/hostile/necro/zombie/crimson)
+
+/mob/living/simple_animal/hostile/necro/zombie/putrid
+	icon_living = "zombie" //The original
+	icon_state = "zombie"
+	icon_dead = "zombie_dead"
+	desc = "A reanimated corpse that looks like it has seen better days. This one appears to be quite gluttenous"
+	maxHealth = 150
+	health = 150
+	can_evolve = 0
+	var/zombify_chance = 25 //Down with hardcoding
+	break_doors = CAN
+
+/mob/living/simple_animal/hostile/necro/zombie/putrid/eat(mob/living/carbon/human/target)
+	..()
+	if(target.health < -150  && isjusthuman(target)) //Gotta be a bit chewed on
+		visible_message("<span class='warning'>\The [target] stirs, as if it's trying to get up.</span>")
+		if(prob(zombify_chance))
+			zombify(target)
+
+/mob/living/simple_animal/hostile/necro/zombie/putrid/proc/zombify(var/mob/living/carbon/human/target)
+	//Make the target drop their stuff, move them into the contents of the zombie so the ghost can at least see how its zombie self is doing
+	target.drop_all()
+	var/mob/living/simple_animal/hostile/necro/zombie/turned/new_zombie = new /mob/living/simple_animal/hostile/necro/zombie/turned(target.loc)
+	new_zombie.name = target.real_name
+	new_zombie.host = target
+	target.loc = null
```

Comment:
But that also means you should have a destroy() proc that drop_host's so the reference gets removed when the zombie gets deleted